src/info/manifest.c

```diff
@@ -383,6 +386,750 @@ manifestNewInternal(void)
+
+            // Make sure the link is valid
+            manifestLinkCheck(buildData.manifest);
+
+            // If the link check was successful but the destination does not exist then check it again to generate an error
+            if (!linkedInfo.exists)
+                storageInfoP(buildData.storagePg, linkPgPath, .followLink = true);
+
+            break;
+        }
+
+        // Skip special files
+        // -------------------------------------------------------------------------------------------------------------------------
+        case storageTypeSpecial:
+        {
+            LOG_WARN("exclude special file '%s/%s' from backup", strPtr(buildData.pgPath), strPtr(info->name));
+            break;
+        }
+    }
+
+    FUNCTION_TEST_RETURN_VOID();
+}
+
+// Regular expression constants
+#define RELATION_EXP                                                "[0-9]+(|_(fsm|vm)){0,1}(\\.[0-9]+){0,1}"
+#define DB_PATH_EXP                                                                                                                \
+    "(" MANIFEST_TARGET_PGDATA "/(" PG_PATH_GLOBAL "|" PG_PATH_BASE "/[0-9]+)|" MANIFEST_TARGET_PGTBLSPC "/[0-9]+/%s/[0-9]+)"
+
+Manifest *
+manifestNewBuild(
+    const Storage *storagePg, unsigned int pgVersion, bool online, const StringList *excludeList)
+{
+    FUNCTION_LOG_BEGIN(logLevelDebug);
+        FUNCTION_LOG_PARAM(STORAGE, storagePg);
+        FUNCTION_LOG_PARAM(UINT, pgVersion);
+        FUNCTION_LOG_PARAM(BOOL, online);
+        FUNCTION_LOG_PARAM(STRING_LIST, excludeList);
+    FUNCTION_LOG_END();
+
+    ASSERT(storagePg != NULL);
+    ASSERT(pgVersion != 0);
+
+    Manifest *this = NULL;
+
+    MEM_CONTEXT_NEW_BEGIN("Manifest")
+    {
+        this = manifestNewInternal();
+        this->info = infoNew(NULL);
+        this->data.pgVersion = pgVersion;
+        this->data.backupOptionOnline = online;
+
+        // Data needed to build the manifest
+        ManifestBuildData buildData =
+        {
+            .manifest = this,
+            .storagePg = storagePg,
+            .tablespaceId = pgTablespaceId(pgVersion),
+            .online = online,
+            .manifestParentName = MANIFEST_TARGET_PGDATA_STR,
+            .manifestWalName = strNewFmt(MANIFEST_TARGET_PGDATA "/pg_%s", strPtr(pgWalName(pgVersion))),
+            .pgPath = storagePathP(storagePg, NULL),
+        };
+
+        // We won't identify db paths for PostgreSQL < 9.0.  This means that temp relations will not be excluded but it doesn't seem
+        // worth supporting this feature on such old versions of PostgreSQL.
+        // -------------------------------------------------------------------------------------------------------------------------
+        if (pgVersion >= PG_VERSION_90)
+        {
+            ASSERT(buildData.tablespaceId != NULL);
+
+            buildData.dbPathExp = regExpNew(
+                strNewFmt("^" DB_PATH_EXP "$", strPtr(buildData.tablespaceId)));
```

Comment:
1.  This regex is being applied to `manifestName`, e.g. `pg_data/base` or `pg_tblspc/...`.
2. I'm not too worried about it because we are only recursing into directories with the exact name (see line 720 for where the path is built) so `PG_9D4_201409291` would never get to the regexp.
3. Reg exp does not require this char to be escaped, but Perl did in some places as you note.  C does not have this restriction so it is simpler not to escape.